pkgs/by-name/az/az-pim-cli/package.nix

```
@@ -20,29 +18,36 @@ buildGoModule (finalAttrs: {
     hash = "sha256-gf4VscHaUr3JtsJO5PAq1nyPeJxIwGPaiH/QdXKpvQ0=";
   };
 
+  patches = [
+    # removes info we don't have from version command
+    ./version-build-info.patch
+  ];
+
   vendorHash = "sha256-PHrpUlAG/PBe3NKUGBQ1U7dCcqkSlErWX2dp9ZPB3+8=";
 
   nativeBuildInputs = [
     installShellFiles
   ];
 
-  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) (
-    let
-      emulator = stdenv.hostPlatform.emulator buildPackages;
-    in
-    ''
-      installShellCompletion --cmd az-pim-cli \
-        --bash <(${emulator} $out/bin/az-pim-cli completion bash) \
-        --fish <(${emulator} $out/bin/az-pim-cli completion fish) \
-        --zsh <(${emulator} $out/bin/az-pim-cli completion zsh)
-    ''
-  );
+  env.CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
+    "-X github.com/netr0m/az-pim-cli/cmd.version=v${finalAttrs.version}"
+  ];
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd az-pim-cli \
+      --bash <($out/bin/az-pim-cli completion bash) \
+      --fish <($out/bin/az-pim-cli completion fish) \
+      --zsh <($out/bin/az-pim-cli completion zsh)
+  '';
 
   passthru = {
     updateScript = nix-update-script { };
     tests.version = testers.testVersion {
-      command = "HOME=$TMPDIR az-pim-cli --version";
-      package = az-pim-cli;
+      command = "HOME=$TMPDIR az-pim-cli version";
+      package = finalAttrs.finalPackage;
```

```suggestion
      package = finalAttrs.finalPackage;
      version = "v${finalAttrs.version}";
```

Version test is still failing.